agents/agents/executor/executor_utils.go

```diff
@@ -51,50 +46,25 @@ func (e Executor) logToSnapshot(log ethTypes.Log, chainID uint32) (*types.Snapsh
 }
 
 // logType determines whether a log is a `Sent` from Origin.sol or `AttestationAccepted` from Destination.sol.
-func (e Executor) logType(log ethTypes.Log, chainID uint32) contractEventType {
-	contractEvent := contractEventType{
-		contractType: other,
-		eventType:    otherEvent,
-	}
+// TODO: Clean with switch case.
+func (e Executor) logType(log ethTypes.Log, chainID uint32) execTypes.ContractType {
```

Comment:
see my comment above about switching on .(type). Might make more sense here. Also, yes. Definitely 100% switch case.

This is a really, really important method (although more so in the notary and the guard then here).

This should be fuzzed to make sure that not only is the eventtpye correct, but it's coming from the correct contract